src/java.base/share/classes/java/security/PEMRecord.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.security;
+
+import sun.security.util.Pem;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * {@code PEMRecord} is a {@link DEREncodable} that represents Privacy-Enhanced
+ * Mail (PEM) data by its type and Base64 form.  {@link PEMDecoder} and
+ * {@link PEMEncoder} use {@code PEMRecord} when representing the data as a
+ * Java API cryptographic object is not desired or there is no other
+ * {@code DEREncodable} for the type.
+ *
+ * <p>Types with Java API representation, such as a {@link PrivateKey},
+ * can return a {@code PEMRecord} when used with
+ * {@linkplain PEMDecoder#decode(String, Class)}. Using {@code PEMRecord} can
+ * be helpful when generating a representation is not desired or when used
+ * with {@code leadingData}.  {@code leadingData} can depend on which
+ * decode() methods is used.
+ *
+ * <p>{@code PEMRecord} may have a null {@code type} and {@code pem} when
+ * {@code PEMDecoder.decode()} methods encounter only non-PEM data and has
+ * reached the end of the stream. If there is PEM data, {@code type} and
+ * {@code pem} will both be non-null. {@code leadingData} may be null if the
+ * input data only contains PEM data. All values can never be null.
```

Comment:
It feels like the 2 paragraphs above should be in the spec of `Decoder::decode`. Here we only need to say what fields can be null.